src/votekit/models.py

```diff
@@ -1,95 +1,292 @@
+        Yield the status (elected, eliminated, remaining) of the candidates in the given round.
+        DataFrame is sorted by current ranking.
+
+        Args:
+            round_number (int, optional): Round number, defaults to -1 which corresponds to
+                the final round.
+        Returns:
+            pd.DataFrame:
+                Data frame displaying candidate, status (elected, eliminated,
+                remaining), and the round their status updated.
+        """
+        if round_number < -1:
+            raise ValueError("round_number must be -1 or non-negative.")
+
+        new_index = [c for s in self.get_ranking(round_number) for c in s]
+
+        if round_number == -1:
+            round_number = len(self)
 
-            elif step >= 0:
-                while self.state.curr_round < step:
-                    self.run_step()
+        candidates = self._profile.get_candidates()
 
-                while self.state.curr_round > step:
-                    if self.state.previous:
-                        self.state = self.state.previous
-                    else:
-                        raise ValueError("Previous state is None type.")
+        status_df = pd.DataFrame(
+            {"Status": ["Remaining"] * len(candidates), "Round": [0] * len(candidates)},
+            index=candidates,
+        )
 
-            return self.state
+        for i in range(round_number):
+            state = self.election_states[i + 1]
+
+            for s in state.elected:
+                for c in s:
+                    status_df.at[c, "Status"] = "Elected"
+                    status_df.at[c, "Round"] = i + 1
+
+            for s in state.eliminated:
+                for c in s:
+                    status_df.at[c, "Status"] = "Eliminated"
+                    status_df.at[c, "Round"] = i + 1
+
+            for s in state.remaining:
+                for c in s:
+                    status_df.at[c, "Round"] = i + 1
+
+        status_df = status_df.reindex(new_index)
+        return status_df
 
     @abstractmethod
-    def run_step(self, *args: Any, **kwargs: Any):
+    def _run_step(
+        self, profile: PreferenceProfile, prev_state: ElectionState, store_states=False
+    ) -> PreferenceProfile:
+        """
+        Run one step of an election from the given profile and previous state.
+
+        Args:
+            profile (PreferenceProfile): Profile of ballots.
+            prev_state (ElectionState): The previous ElectionState.
+            store_states (bool, optional): True if `self.election_states` should be updated with the
+                ElectionState generated by this round. This should only be True when used by
+                `self._run_election()`. Defaults to False.
+
+        Returns:
+            PreferenceProfile: The profile of ballots after the round is completed.
+        """
         pass
 
     @abstractmethod
-    def run_election(self, *args: Any, **kwargs: Any):
+    def _next_round(self) -> bool:
```

Comment:
It would be good to change the name of this. The name `_next_round` makes me think that it returns the next round of the election. Maybe something like `_is_finished` would be better (of course, this name suggests that you have to invert the logic for things)